WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductSelectorView.swift

```diff
@@ -85,17 +85,11 @@ struct ProductSelectorView: View {
 
     var body: some View {
         VStack(spacing: 0) {
-            if ServiceLocator.featureFlagService.isFeatureFlagEnabled(.sideBySideViewForOrderForm) &&
-                horizontalSizeClass == .regular {
-                productSelectorHeaderTitleRow
-                productSelectorHeaderSearchRow
-                    .padding(.bottom, Constants.defaultPadding)
-                    .background(Color(.listForeground(modal: false)))
+            if ServiceLocator.featureFlagService.isFeatureFlagEnabled(.sideBySideViewForOrderForm) {
+                productSelectorHeader
             } else {
-                productSelectorHeaderSearchRow
-                productSelectorHeaderTitleRow
+                legacyProductSelectorHeader
```

Comment:
> Observe that the header of the product selector has a grey background, with the search bar above the selection/filter buttons

Thanks for addressing the search button arrangement, that slipped through. I think there's a mistake with the grey background thought, isn't it? We always should be using the `.background(Color(.listForeground(modal: false)))` for both the legacy and the new headers, this is not grey in previous versions, neither in the designs.

```suggestion
                legacyProductSelectorHeader
                .background(Color(.listForeground(modal: false)))
```

Reply:
No, it's grey in previous builds, that's what tipped me off to the issue.

Here's a screenshot from 16.5 for comparison with something well before we touched it (but it's like this right up to 17.4.)

![16.5 product selector](https://github.com/woocommerce/woocommerce-ios/assets/2472348/83501223-7057-4705-85a4-ea1e58e5bd8b)

Reply:
Interesting, thanks for sharing the screenshot. I checked previous builds but not as far as that one 🤔

Reply:
How did you check them? I ask because the last release had it as grey, so it's not that far back.

When you need to do this, I'd recommend using TestFlight, as it's probably the quickest way to grab a release build for a specific version number, and you can be quite confident you've got the right one.

Reply:
> How did you check them? I ask because the last release had it as grey, so it's not that far back.

I checked 17.4. through TestFlight, which shows this view using `. listForeground` rather than grey. Along with the designs is what made me thought it shouldn't be grey. 

I must have loaded 17.5 as well, but makes no much sense since would include the changes that we're attempting to solve with this PR. I loaded now 17.3.0.1 and I see that's greyed out as you shared in the screenshot 👍